credentials/generate-revocation-set.py

```diff
@@ -272,57 +350,68 @@ def get_revocation_points(self) -> list[dict]:
 
         Parameters
         ----------
-        cert: x509.Certificate
-            Certificate
+        issuer_subject_key_id: str
+            Subject key ID
 
         Returns
         -------
-        str
-            Issuer certificate in PEM format
+        list[dict]
+            List of revocation points
         '''
-        issuer_name_b64 = get_issuer_b64(cert)
-        akid = get_akid(cert)
-        if akid is None:
-            return
 
-        # Convert CRL Signer AKID to colon separated hex
-        akid_hex = akid.hex().upper()
-        akid_hex = ':'.join([akid_hex[i:i+2] for i in range(0, len(akid_hex), 2)])
+        response = self.get_dcld_cmd_output_json(['query', 'pki', 'revocation-points',
+                                                  '--issuer-subject-key-id', issuer_subject_key_id])
+        logging.debug(f"Response revocation points: {response}")
+        return response["pkiRevocationDistributionPointsByIssuerSubjectKeyID"]["points"]
 
+    def get_certificate(self, issuer_name_b64: str, akid_hex: str) -> tuple[bool, Optional[x509.Certificate]]:
+        '''
+        Get certificate from DCL
+        '''
+        query_cmd_list = ['query', 'pki', 'x509-cert', '-u', issuer_name_b64, '-k', akid_hex]
         logging.debug(
-            f"Fetching issuer from:{self.rest_node_url}/dcl/pki/certificates/{issuer_name_b64}/{akid_hex}")
+            f"Fetching issuer from dcl query{' '.join(query_cmd_list)}")
+        response = self.get_dcld_cmd_output_json(query_cmd_list)
+        return self.get_first_approved_certificate(response, akid_hex)
 
-        if self.use_rest:
-            response = requests.get(
-                f"{self.rest_node_url}/dcl/pki/certificates/{issuer_name_b64}/{akid_hex}").json()
-        else:
-            response = self.get_dcld_cmd_output_json(
-                ['query', 'pki', 'x509-cert', '-u', issuer_name_b64, '-k', akid_hex])
 
-        issuer_certificate = response["approvedCertificates"]["certs"][0]["pemCert"]
+class RESTDCLDClient(DCLDClientInterface):
+    '''
+    A client for interacting with DCLD using the REST API.
 
-        logging.debug(f"issuer: {issuer_certificate}")
+    '''
 
-        try:
-            issuer_certificate_object = x509.load_pem_x509_certificate(bytes(issuer_certificate, 'utf-8'))
-        except Exception:
-            logging.error('Failed to parse PAA certificate')
-            return
+    def __init__(self, rest_node_url: str):
+        '''
+        Initialize the client
+
+        rest_node_url: str
+            RESTful API URL
+        '''
 
-        return issuer_certificate_object
+        self.rest_node_url = rest_node_url
+
+    def get_revocation_points(self) -> list[dict]:
+        '''
+        Get revocation points from DCL
+
+        Returns
+        -------
+        list[dict]
+            List of revocation points
+        '''
+
+        response = requests.get(f"{self.rest_node_url}/dcl/pki/revocation-points").json()
```

Comment:
Should we handle http request/response exceptions? For example, http client can timeout, or it may return some non-expected HTTP errors etc. WDYT?

Reply:
I've added send_get_request to the base class that will handle errors from request.get calls for json objects.

Reply:
SG.